doc/usage/restructuredtext/domains.rst

```diff
@@ -253,6 +253,22 @@ The following directives are provided for module and class contents:
 
       .. versionadded:: 2.4
 
+.. rst:directive:: .. py:property:: name
+
+   Describes an object property.
+
+   .. versionadded:: 3.0
+
+   .. rubric:: options
+
+   .. rst:directive:option:: abstractmethod
+      :type: no value
+
+      Indicate the property is an abstract method.
+
+   .. rst:directive:option:: type: type of the property
+      :type: text
```

Comment:
Maybe we should also add the type the setter accepts? E.g. a `path` property could accept any `typing.PathLike` but always coerces it to `pathlib.Path`.

Reply:
Not sure how you'd want to write or display that. Perhaps the folks over at mypy can recommend a spelling?

Reply:
In code it’s obvious to express this: A return type annotation for the getter and a parameter type annotation on the setter.

In rST it could be `:get-type:` or `:type:`, and if a setter exists, also `:set-type:`.

I don’t know a concise way to display it (like “**Return Type**” for functions), except for spelling out “can be set to $Type”

Reply:
Perhaps if `typing.Descriptor` could be added, as was suggested in the comment thread 
 at https://github.com/python/mypy/issues/244#issuecomment-620113658, then this could just be written `:type: Descriptor[GetType, SetType]`.

Reply:
I think this should be an another topic. The goal of this PR is replacing property-description to `py:property` directive.